java.source.base/src/org/netbeans/api/java/source/TreeUtilities.java

```diff
@@ -1927,6 +1927,23 @@ public boolean isPartOfCompoundVariableDeclaration(@NonNull Tree tree) {
         return false;
     }
 
+    /**
+     * Check the tree has compile error in given collection of errors.
+     *
+     * @param tree : compilation tree
+     * @param errors : Collection of error code
+     * @return true if tree has compile error present in collection of errors.
+     * @since 2.37.0
+     */
+    public boolean isTreeHasError(Tree tree, Collection<String> errors) {
```

Comment:
`isTreeHasError` contains 2 verbs :) maybe `hasTreeError` could be better name (no need to follow BeanPattern conventions here). 

Please consider to extend the behaviour for empty collection, in the sense of "any": any error on the Tree would then result in `true`.

Because of the usage, I would also recommend to use `String...` vararg parameter, since the caller can directly use constants, IMHO typically constant fixed number of errors will be checked this way. The collection creation could not be entirely avoided, since `Arrays.asList` and `indexOf` could be used.

Reply:
add varArgs parameter and change the API name to hasError()